lc-run: Run compiled Builder bytecode files without engine

peter-b opened the pull request:
```
$ lc-run --help
Usage: lc-run [OPTIONS] [--] LCMFILE [ARGS ...]

Run a compiled Modular Livecode bytecode file.

Options:
  -H, --handler NAME   Specify name of handler to run.
  --list-handlers      List possible entry points in LCMFILE and exit.
  -h, --help           Print this message.
  --                   Treat next argument as bytecode filename.

Any ARGS are available in "the command arguments".

Report bugs to <http://quality.runrev.com/>
```

peter-b committed:
[[feature-lcrun]] Add basic options to lc-run.

Add some basic options to lc-run:

* -h, --help (print usage)
* -- (end options)

peter-b committed:
[[feature-lcrun]] lc-run: Support for arguments with options.

peter-b committed:
[[feature-lcrun]] lc-run: Add -H, --handler option for specifying entry point.

This allows specifying a top-level handler other than "main".

peter-b committed:
[[feature-lcrun]] lc-run: Add --list-handlers mode.

Add a mode which lists all public handlers in the target module file
that can be invoked via `lc-run --handler <name>`.

peter-b committed:
[[feature-lcrun]] Add lc-run program.

Add a new program that runs LiveCode bytecode (`.lcm`) files without
the engine.  It runs a bytecode file iff the file contains a `main`
handler matching:

    public handler main() as undefined
    	/* Body */
    end handler

If the handler completes, `lc-run` exits with status 0.  If an
uncaught error occurs, `lc-run` prints a message and exits with status
126.

runrevmark commented on toolchain/lc-compile/src/lc-run.cpp:
```diff
@@ -0,0 +1,426 @@
+		MCRunStartupError();
+
+	/* ---------- Start VM */
+	MCAutoValueRefBase<MCScriptModuleRef> t_module;
```
MCScript*Ref are not currently ValueRef's so you can't use MCAutoValueRefBase.

peter-b replied:
Should I add a suitable auto type, or add explicit cleanup?

peter-b committed:
[[feature-lcrun]] libscript: Add auto cleanup types.

peter-b committed:
[[feature-lcrun]] Use libscript auto types.

libscript's types aren't based on MCValueRef, so it's invalid to use
an MCAutoValueRefBase template with them.

peter-b commented:
Hi @runrevmark, that should be fixed now; I've added `MCAutoScriptModuleRef` and `MCAutoScriptInstanceRef` and used them instead.

runrevmark commented:
Reviewed-by: @runrevmark.
